app/models/test.rb

```diff
@@ -4,11 +4,18 @@ class Test < ApplicationRecord
   has_many :users, through: :results
   belongs_to :category
   belongs_to :authored_tests, class_name: 'User', foreign_key: 'user_id'
+  validates :title, presence: true,
+                    uniqueness: { scope: :level, message: :uniq_title_with_level }
+  validates :level, numericality: { greater_than_or_equal_to: 0 }
```

Comment:
ТЗ: "Уровень Теста может быть только положительным целым числом"
обрати внимание на [документацию](https://guides.rubyonrails.org/active_record_validations.html#numericality) к хелперу `numericality`
> By default, it will match an optional sign followed by an integral or floating point number.

целочисленность не гарантирована по умолчанию - надо еще что-то добавить